extensions/amp-cxense-player/0.1/amp-cxense-player.js

```diff
@@ -0,0 +1,319 @@
+    }
+
+    /** @private */
+    postMessage_(data) {
+      data = extend(true, {
+        location: location,
+        width: this.getWin().offsetWidth,
+        height: this.getWin().offsetHeight,
+      }, data || {});
+
+      return this.iframe_.contentWindow.postMessage(
+            JSON.stringify(data), cxDefaults.embedHost
+        );
+    }
+
+    /** @private */
+    buildWidgetPlaceholder_() {
+      const doc = this.getDoc_();
+
+      this.placeholder_ = doc.createElement('div');
+      this.placeholder_.className = 'amp-cxense-player-placeholder';
+
+      const spinner = doc.createElement('div');
+      spinner.className = 'amp-cxense-player-loader';
+      spinner.appendChild(doc.createElement('div'));
+      spinner.appendChild(doc.createElement('div'));
+      spinner.appendChild(doc.createElement('div'));
+
+      this.placeholder_.appendChild(spinner);
+      this.element.appendChild(this.placeholder_);
+      this.applyFillContent(this.placeholder_);
+    }
+
+    /** @private */
+    getDoc_() {
+      return this.getWin().document;
+    }
+}
+
+AMP.registerElement('amp-cxense-player', AmpCxense, CSS);
+
+function attrHash(el, prefix) {
+  const ret = {};
+  const attrs = el.attributes;
+  const p = prefix || '';
+  forEach(attrs, function(i, node) {
+    if (node.nodeName.indexOf(p + '-') == 0) {
+      const name = dashToCamelCase(node.nodeName.slice(p.length + 1));
+      ret[name] = resolveType(node.nodeValue);
+    }
+  });
+  return ret;
+}
+
+function forEach(arr, callback) {
+  const a = arr || [];
+  let i;
+  const l = a.length;
+  let ret;
+  if (a instanceof Array || a.length >= 0) {
+    for (i = 0;i < l;i++) {
+      ret = callback(i,a[i]);
+      if (ret == false) {
+        break;
+      }
+    }
+  }
+    else if (a instanceof Object) {
+      for (i in a) {
+        if (!a.hasOwnProperty(i)) {
+          continue;
+        }
+        ret = callback(i,a[i]);
+        if (ret == false) {
+          break;
+        }
+      }
+    }
+}
+
+function queryString(obj) {
+  const parts = [];
+  if (!obj) {
+    return '';
+  }
+  forEach(obj, function(key, val) {
+    if (val != null) {
+      parts.push(encodeURIComponent(key) + '=' + encodeURIComponent(val));
+    }
+  });
+  return parts.join('&');
+}
+
+// expands an object with dot notation keys into a multi-level tree
+function expandKeys(obj) {
+  const ret = {};
+  forEach(obj, function(key, val) {
+    expandKey(key, val, ret);
+  });
+  return ret;
+}
+
+// expands a dot notation key into the given object
+function expandKey(key, value, parent) {
+  const parts = key.split('.');
+  if (!parent) {
+    parent = {};
+  }
+  let node = parent;
+  let k;
+  while (k = parts.shift()) {
+    if (parts.length == 0) {
+      node[k] = value;
+    } else if (typeof node[k] == 'object') {
+      node = node[k];
+    } else {
+      node = node[k] = {};
+    }
+  }
+  return parent;
+}
+
+function recurseObject(object, onKeypathFn, keypathPrefix) {
```

Comment:
What do you need this for? Your example doesn't appear to have any recursive stuff.